src/analysis/calculus/uniform_limits_deriv.lean

```diff
@@ -0,0 +1,394 @@
+  have : 0 < (2⁻¹ * r⁻¹ * ε),
+  { exact mul_pos (mul_pos (by norm_num) (by simp [hrpos])) hε.lt, },
+  specialize foo (2⁻¹ * r⁻¹ * ε) this.gt,
+  cases foo with N2 hN2,
+
+  -- Some annoying manipulation
+  let N := max N1 N2,
+  refine ⟨N, λ m hm n hn y hy, _⟩,
+  rw dist_eq_norm,
+
+  -- Apply the triangle inequality
+  have : f m y - f n y = (f m y - f n y) - (f m x - f n x) + (f m x - f n x),
+  { rw sub_add_cancel, },
+  rw this,
+  have : ∥f m y - f n y - (f m x - f n x) + (f m x - f n x)∥ ≤
+    ∥f m y - f n y - (f m x - f n x)∥ + ∥f m x - f n x∥,
+  { exact norm_add_le _ _, },
+  refine lt_of_le_of_lt this _,
+
+  -- The case y = x is trivial and causes some divide by zero errors throughout the
+  -- proof, so we just take care of it now
+  by_cases hyxx : y = x,
+  { simp [hyxx],
+    rw ←dist_eq_norm,
+    have := hN1 m
+      (le_trans (le_max_left N1 N2) hm.le) n (le_trans (le_max_left N1 N2) hn.le),
+    transitivity,
+    exact this,
+    rw mul_lt_iff_lt_one_left hε.lt,
+    norm_num, },
+
+  -- Conveniences that the ring solver can't figure out on its own
+  have hxyy : y - x ≠ 0, exact λ H, hyxx (sub_eq_zero.mp H),
+  have hxyy' : ∥y - x∥ ≠ 0, simp [hxyy],
+
+  -- Multiply and divide by the difference quotient denominator
+  have : ∥f m y - f n y - (f m x - f n x)∥ =
+    ∥y - x∥ * (∥y - x∥⁻¹ * ∥f m y - f n y - (f m x - f n x)∥),
+  { exact (mul_inv_cancel_left₀ hxyy' _).symm, },
+  rw this,
+
+  specialize hN2 m (ge_trans hm (by simp)) n (ge_trans hn (by simp)),
+  have : ∀ (x_1 : E), x_1 ∈ closed_ball x r → ∥ f' m x_1 - f' n x_1∥ ≤ 2⁻¹ * r⁻¹ * ε,
+  { intros y hy,
+    rw ←dist_eq_norm,
+    exact (hN2 y hy).le, },
+  have hxb : x ∈ closed_ball x r, simp [hrpos.le],
+  have mvt := mean_value_theorem_for_differences hrR (hf m) (hf n) this hy hxb,
+  specialize hN1 m (ge_trans hm (by simp)) n (ge_trans hn (by simp)),
+  rw dist_eq_norm at hN1,
+
+  have : ε = (2⁻¹ * ε) + (2⁻¹ * ε), ring,
+  rw this,
+  have : r⁻¹ * r = 1, { exact inv_mul_cancel hrpos.ne.symm, },
+
+  have : ∥y - x∥ * (∥y - x∥⁻¹ * ∥f m y - f n y - (f m x - f n x)∥) ≤ 2⁻¹ * ε,
+  { have : ∥y - x∥ ≤ r, { rw [mem_closed_ball, dist_eq_norm] at hy, exact hy, },
+    calc ∥y - x∥ * (∥y - x∥⁻¹ * ∥f m y - f n y - (f m x - f n x)∥) ≤ r * (2⁻¹ * r⁻¹ * ε) :
+      mul_le_mul this mvt (mul_nonneg (by simp) (by simp)) (hrpos.le)
+    ... = 2⁻¹ * ε : begin
+      ring_nf,
+      rw [mul_assoc, inv_mul_cancel hrpos.ne.symm],
+      ring,
+    end },
+  exact add_lt_add_of_le_of_lt this hN1,
+end
+
+/-- (d/dx) lim_{n → ∞} f_n x = lim_{n → ∞} f'_n x on a closed ball when the f'_n
+converge _uniformly_ to their limit. -/
+lemma has_fderiv_at_of_tendsto_uniformly_on
+  (hrR : r < R)
+  (hf : ∀ (n : ℕ), ∀ (y : E), y ∈ ball x R → has_fderiv_at (f n) (f' n y) y)
+  (hfg : ∀ (y : E), y ∈ closed_ball x r → tendsto (λ n, f n y) at_top (nhds (g y)))
+  (hfg' : tendsto_uniformly_on f' g' at_top (closed_ball x r)) :
+  ∀ y : E, y ∈ ball x r → has_fderiv_at g (g' y) y :=
```

Comment:
What are the weakest conditions to conclude `has_fderiv_at g (g' y) y` for a single value `y`? That would be the more convenient lemma to apply in applications, and should be the statement of this lemma. 
Do you need `hf` only at `y` or on a neighborhood around `y`? In the latter case, use something like `∀ n, ∀ᶠ x in 𝓝 y, has_fderiv_at (f n) (f' n x) x`.

Reply:
I think I responded to this comment below!